src/gui/shareusergroupwidget.cpp

```diff
@@ -247,17 +249,17 @@ void ShareUserGroupWidget::slotShareCreated(const QSharedPointer<Share> &share)
 
 void ShareUserGroupWidget::slotSharesFetched(const QList<QSharedPointer<Share>> &shares)
 {
-    QScrollArea *scrollArea = _parentScrollArea;
-
-    auto newViewPort = new QWidget(scrollArea);
-    auto layout = new QVBoxLayout(newViewPort);
-    layout->setContentsMargins(0, 0, 0, 0);
     int x = 0;
-    int height = 0;
     QList<QString> linkOwners({});
-
+    
     ShareUserLine *justCreatedShareThatNeedsPassword = nullptr;
-
+       
+    QLayoutItem *shareUserLine;
+    while ((shareUserLine = _shareUserGroup->takeAt(0)) != nullptr) {
```

Comment:
@camilasan You may make it shorter and fix clang-tidy complaints:

`while ((shareUserLine = _shareUserGroup->takeAt(0)) != nullptr)` -> `while (QLayoutItem *shareUserLine = _shareUserGroup->takeAt(0))`